zyte_common_items/pages.py

```diff
@@ -54,6 +61,33 @@ def _get_metadata_class(cls: type) -> Optional[Type[MetadataT]]:
     return get_generic_param(cls, HasMetadata)
 
 
+class PriceMixin(FieldsMixin):
+    """Provides price-related field implementations."""
+
+    _parsed_price: Optional[Price] = None
+
+    async def _get_parsed_price(self):
+        if self._parsed_price is None:
+            price_field = getattr(self, "price", None)
+            if price_field is None:
+                return None
```

Comment:
Does it mean that if `price` field returns None, this None result is not cached, and price extraction will be called multiple times?

Reply:
It seems worth doing, I don't see downsides in using a sentinel here.